packages/discord.js-utilities/src/lib/utilities.ts

```diff
@@ -0,0 +1,47 @@
+import { Permissions } from 'discord.js';
+import { isGuildBasedChannel } from './type-guards';
+import type { TextBasedChannelTypes } from './utility-types';
+
+const canReadMessagesPermissions = new Permissions(['VIEW_CHANNEL']);
+
+/**
+ * Determines whether or not we can send messages in a given channel.
+ * @param channel The channel to test the permissions from.
+ * @returns Whether or not we can send messages in the specified channel.
+ */
+export function canReadMessages(channel: TextBasedChannelTypes): boolean {
+	return isGuildBasedChannel(channel) ? channel.permissionsFor(channel.guild.me!)!.has(canReadMessagesPermissions) : true;
+}
```

Comment:
Instead of writing several functions of similar structure, perhaps use a similar approach to [what I suggested on Kyra's PR](https://github.com/sapphiredev/utilities/pull/118#discussion_r683066642).